pandas/core/strings.py

```diff
@@ -1743,13 +1815,17 @@ def str_get(arr, i):
     """
 
     def f(x):
-        if isinstance(x, dict):
-            return x.get(i)
-        elif len(x) > i >= -len(x):
-            return x[i]
+        try:
```

Comment:
Previously, this was handled inside `_map` by

1. calling the `f` on each element inside a big try/except
2. If the callable failed, we'd define a new callable that wrapped `f` inside a try / except, and mapped the new callable over the values.

That means something like `Series(['a'] * 1000 + [None]).str.get(0)` would call this `f` 2002 times! The first 1,000 were good, then we failed on 1,001. So we define a new wrapped callable and call that 1,001 more times.

Better to do the wrapping here.